spec/v1.0-draft/resource_descriptor.md

```diff
@@ -0,0 +1,168 @@
+# ResourceDescriptor field type specification
+
+Version: v1.0-draft
+
+A size-efficient description of any software artifact or resource (mutable
+or immutable).
+
+## Schema
+
+```json
+{
+  "name": "<NAME>",
+  "uri": "<RESOURCE URI>",
+  "digest": { "<ALGORITHM>": "<HEX VALUE>", ... },
+  "content": "<BASE64 VALUE>",
+  "downloadLocation": "<RESOURCE URI>"
+  "mediaType": "<MIME TYPE>",
+  "annotations": {
+    "<FIELD_1>": { /* object */ },
+    "<FIELD_2>": { /* object */ },
+    ...
+  }
+}
+```
+
+## Fields
+
+Though all fields are optional, a ResourceDescriptor MUST specify one of
+`uri` or `digest` at a minimum.
+
+`name` _string, optional_
+
+> Machine-readable identifier for distinguishing between descriptors.
+>
+> The semantics are up to the producer and consumer. The `name` SHOULD be
+> stable, such as a filename, to allow consumers to reliably use the `name`
+> as part of their policy.
+
+`uri` _[ResourceURI], optional_
+
+> A URI used to identify the resource or artifact globally.
+>
+> When a `digest`cannot be computed for the resource or artifact, the
+> producer MUST set this field.
+
+`digest` _[DigestSet], optional_
+
+> A set of cryptographic digests of the contents of the resource or artifact.
+>
+> When known, the producer SHOULD set this field to denote an immutable
+> artifact or resource. The producer and consumer must agree on acceptable
+> algorithms.
```

Comment:
I've updated the Semantics section in a way that will hopefully clarify how to handle `digest`. I'm inclined to not put such a strong requirement on the `digest` and `content` combination, because I think there might be cases where `uri`, `digest` and `content` are all set, and somehow the `digest` is computed over the `uri` rather than the `content`. In the TektonTask example, for instance, it's not immediately clear if the report pointed to by the `uri` is a larger artifact that has the `digest` and the `content` just contains a summary of the larger report.

>  a consumer MAY compute an arbitrary digest over content and consider that equivalent to being set here

Is this referring to a `digest` field _within_ the `content`?

Reply:
Let's not block this PR on that. I'll might send a follow-up PR with a proposed tweak for the wording, but this is good enough for now.

Reply:
Thanks!